project_euler/problem_066/sol1.py

```diff
@@ -0,0 +1,132 @@
+"""
+
+
+Project Euler 66
+https://projecteuler.net/problem=66
+https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Algorithm
+
+
+Consider quadratic Diophantine equations of the form:
+
+x^2 – Dy^2 = 1
+
+For example, when D=13, the minimal
+solution in x is 6492 – 13×1802 = 1.
+
+It can be assumed that there are no solutions in
+positive integers when D is square.
+
+By finding minimal solutions in x for
+D = {2, 3, 5, 6, 7}, we obtain the following:
+
+3^2 – 2×2^2 = 1
+2^2 – 3×1^2 = 1
+9^2 – 5×4^2 = 1
+5^2 – 6×2^2 = 1
+8^2 – 7×3^2 = 1
+
+Hence, by considering minimal solutions in x for D ≤ 7,
+the largest x is obtained when D=5.
+
+Find the value of D ≤ 1000 in minimal solutions of
+x for which the largest value of x is obtained.
+
+"""
+
+from math import sqrt
+
+
+def continued_fraction(n: int) -> list:
+
+    """
+    function to find continued fraction
+
+    >>> continued_fraction(2)
+    [1]
+
+    >>> continued_fraction(3)
+    [1, 1]
+
+    >>> continued_fraction(5)
+    [2]
+
+    >>> continued_fraction(6)
+    [2, 2]
+
+    """
+    mn = 0.0
+    dn = 1.0
+    a0 = int(sqrt(n))
+    an = int(sqrt(n))
+    convergents = [a0]
+    # period = 0
+    if a0 != sqrt(n):
+        while an != 2 * a0:
+            mn = dn * an - mn
+            dn = (n - mn ** 2) / dn
+            an = int((a0 + mn) / dn)
+            convergents.append(an)
+    return convergents[:-1]
+
+
+def simple_frac(cf: list()) -> int:
```

Comment:
Descriptive function and parameter name.
Proper type hints needed: https://docs.python.org/3/library/typing.html